### DISCUSSION TITLE
Обход блокировки по IP

### DISCUSSION THREAD

**Discussion by NerdzzyDev** [bol-van/zapret#338] [Show and tell]
Написал небоьшую инструкцию, как в моем случае обошел блокировку по IP для instagram.com 
В моём случае, выполнил следующие шаги:

### Шаги:

1. **Проверка доступности  Instagram.com через `ping`:**
   - В  терминале выполните команду:
     ```bash
     ping instagram.com
     ```
   - Если нет ответа от сервера, значит доступ к IP-адресу Instagram заблокирован.

2. **Поиск IP-адреса Instagram с помощью `nslookup`:**
   - Используйте команду `nslookup`, чтобы получить IP-адреса через DNS-сервер Google:
     ```bash
     nslookup instagram.com 8.8.8.8
     ```

   - Проверьте полученные IP-адреса с помощью команды `ping`:
     ```bash
     ping <IP-адрес>
     ```
   - Если нет ответа от сервера, значит можно попробовать другие DNS, например 1.1.1.1 - если и это не помогло, то переходим к следующему шагу.

3. **Использование VPN для получения рабочего IP-адреса:**
   - Включите VPN на телефоне или компьютере.
   - Выполните `ping` для домена Instagram:

     ```bash
     ping instagram.com
     ```
   - В случае, если ответ от сервера получен, обратите внимание на IP-адрес в выводе, в моем случае это был `157.240.9.174` - стоит проверить его без подключенного vpn
     ```bash
     ping 157.240.9.174
     ```
В случае успешного ответа от сервера, переходим к следующему шагу. Для сайтов расположенных за cloudflare или имеющих всего 1 ip адресс данная стратегия не подойдет. В этом случае поможет только использования VPN или proxy.

4. **Редактирование файла `hosts`:**
   - Откройте файл `hosts` для редактирования с правами суперпользователя:
     ```bash
     sudo vim /etc/hosts
     ```
   - Добавьте строки с найденным IP-адресом:
     ```
     157.240.9.174 instagram.com www.instagram.com
     ```
   - Сохраните изменения и выйдите из редактора.

5. Сбрасываем кеш DNS 
На Windows:
```bash
ipconfig /flushdns
```
на роутере:
```bash
/etc/init.d/dnsmasq restart
```

6. **Проверка доступа к Instagram:**
   - Откройте браузер и попробуйте зайти на сайт Instagram.
   - Если доступ восстановлен, значит настройка успешна.

### Кратко:
Проверяем доступность Instagram через `ping`, находим рабочий IP-адрес с помощью VPN или DNS, и затем добавляем его в файл `hosts` для обхода блокировки.
Пример ping 
![IMG_20240829_134622_367.jpg](https://github.com/user-attachments/assets/c698a21c-35b4-4e27-a87e-af1995aff1b0)

Также, можно и с другими сайтами, надеюсь помог :)

P.s. (Вместо vim можно использовать любой другой редактор, на всякий случай, чтобы редактировать в VIM - нажмимаем `i` после чего добавляем нужную строку, затем нажимаем  `ESC` и ввыодим `:wq` после чего `Enter`, чтобы проверить что все сделали верно 
`cat /etc/hosts`

UPD. Спасибо @bol-van  за советы по исправлению ошибок, теперь иснтрукция более универсальна и без лишних пакетов.

**Comment by numbereleven-a** [comment]
посмотрел ip через вкл vpn, на роутере стало пинговаться
с компа нет
с телефона нет

**Reply by NerdzzyDev** [reply]
При этом добавили запись в hosts лист? Чтобы обращение шло к нему, как в примере?

**Reply by numbereleven-a** [reply]
> При этом добавили запись в hosts лист? Чтобы обращение шло к нему, как в примере?
да запись появилась, роутер стали пинговать айпишник через имя сайта
а вот винда и iphone нет
я на опенврт это делал, может оно и не должно работать после этого на винде)

**Reply by NerdzzyDev** [reply]
> > При этом добавили запись в hosts лист? Чтобы обращение шло к нему, как в примере?
> да запись появилась, роутер стали пинговать айпишник через имя сайта
> а вот винда и iphone нет
> я на опенврт это делал, может оно и не должно работать после этого на винде)

---
Должно, если настроить DNS на роутере - чтобы все устройства, подключенные к роутеру использовали правила, которые установлены на нем.

Текстом описывать долго, но вот видео пример, как это можно сделать на openwrt:

https://youtu.be/6prF6j7c9Us

**Reply by bol-van** [reply]
DNS кэши на клиентах надо сбросить
на винде ipconfig /flushdns
на рутере /etc/init.d/dnsmasq restart

**Reply by NerdzzyDev** [reply]
> DNS кэши на клиентах надо сбросить
> на винде ipconfig /flushdns
> на рутере /etc/init.d/dnsmasq restart

---
Хотел узнать, а сама инструкция нормальной вышла? 😅 Или есть какие-либо ошибки и неточности?

**Reply by bol-van** [reply]
ping не устанавливает никаких "соединений". он отсылает пакеты icmp echo request и получает icmp echo reply.
Корректней было бы говорить "есть ответ от сервера" или "нет ответа от сервера'.

dig - это часть самого известного DNS сервера bind9
Достаточно обьемный пакет, хотя обычно его разбивают на части в дистрибутивах.
bind9-dnsutils в debian. bind-dig в openwrt.
Нигде его обычно нет по умолчанию, а в openwrt он сильно сьест флэш, если нет extroot.
nslookup решает ту же самую задачу и присутствует везде из коробки

**Reply by NerdzzyDev** [reply]
> ping не устанавливает никаких "соединений". он отсылает пакеты icmp echo request и получает icmp echo reply.
> Корректней было бы говорить "есть ответ от сервера" или "нет ответа от сервера'.
> 
> dig - это часть самого известного DNS сервера bind9
> Достаточно обьемный пакет, хотя обычно его разбивают на части в дистрибутивах.
> bind9-dnsutils в debian. bind-dig в openwrt.
> Нигде его обычно нет по умолчанию, а в openwrt он сильно сьест флэш, если нет extroot.
> nslookup решает ту же самую задачу и присутствует везде из коробки

---
Спасибо, сейчас все исправлю!

**Reply by numbereleven-a** [reply]
после чистки кэшей работает

**Reply by numbereleven-a** [reply]
правда видео не играются зато заходится) в целом пофиг, через прилу работает кое как, просто интересно было
#+html: <details><summary>jpg</summary>
![0a7fcf4ce86618d640976cf7fb78e00b](https://github.com/user-attachments/assets/ab5e1c4c-6d56-4ac6-b188-0b7d708014ac)
![21814b5f4889de7561d092726d952ee2](https://github.com/user-attachments/assets/bd2e8c4f-2fa8-4d6f-a262-51c091505603)
![8afa5494700049b745bf252d43c7c0a3](https://github.com/user-attachments/assets/eec2eba3-e0b2-48fd-95d9-4365c5d2cfc1)
#+html: </details>

**Reply by NerdzzyDev** [reply]
> правда видео не играются зато заходится)

--- 
Спасибо за фото!
Тут два варианта, или подождать немного (dns записи еще могли не обновится)

Или делаем тоже самое, но с поддоменамами :

```bash
ping www.instagram.com
ping graph.instagram.com
ping api.instagram.com
ping cdn.instagram.com
```

Если какой-то из них заблочен, то делаем по инструкции, по итогу все будет работать быстро и без тормозов, будто не блокировали :)

**Reply by numbereleven-a** [reply]
> > правда видео не играются зато заходится)
> 
> Если не секрет, то какой сайт разблокировали?

instagram

**Comment by electrifying** [comment]
> Интересует, есть ли какие-то ошибки

Ну, можно было сразу написать что инструкция только для линукса. Ибо команды dig не является виндовой. К тому же это походу работает только для крупных сайтов для torrentgalaxy.to,например, находит только один ip

**Reply by NerdzzyDev** [reply]
Спасибо, важное уточнение! Да, действительно, если у сайта всего 1 IP, то данный способ не подойдёт. А также в том случае, если сайт находится за cloudflare, т.к. не получить прямой IP.

Для Windows можно использовать:

```bash
nslookup instagram.com 8.8.8.8
```

**Comment by recultivator** [comment]
Сделал всё по инструкции, в итоге в файле такие записи: 

`157.240.214.174 instagram.com www.instagram.com
157.240.221.63 graph.instagram.com
157.240.214.63 api.instagram.com  instagram.c10r.instagram.com
157.240.205.63 
163.70.147.63 static.cdninstagram.com scontent.cdninstagram.com
`
но теперь сайт-то открывается, а видео и фото при этом не загружаются.
Куда копать?

**Reply by NerdzzyDev** [reply]
Не открываются с телефона или с пк?

По хорошему, чтобы понять что именно блокируют можно посмотреть логи или запросы в консоли браузера. ( [Как тут]()https://github.com/bol-van/zapret/discussions/338#discussioncomment-10502592)

Для телефона логи отследить будет сложнее, можно использовать tcpdump и лог --debug от nfqws

**Reply by recultivator** [reply]
> Не открываются с телефона или с пк?
> 
> По хорошему, чтобы понять что именно блокируют можно посмотреть логи или запросы в консоли браузера. ( Как тут[#338 (reply in thread)](https://github.com/bol-van/zapret/discussions/338#discussioncomment-10502592))
> 
> Для телефона логи отследить будет сложнее, можно использовать tcpdump и лог --debug от nfqws


И там, и там не открываются. В какой-то момент с телефона всё заработало на пару минут, но так и не смог уловить с чем это было связано. Неужели у меня только так
В консоли такого рода ошибки показывает:
![image](https://github.com/user-attachments/assets/1cb39f77-7017-4090-9a41-8c9559de007f)

**Reply by M-A-M-B-A** [reply]
Поддерживаю вопрос. Как через DOH получить список доступных по ping ip.

**Reply by numbereleven-a** [reply]
> Поддерживаю вопрос. Как через DOH получить список доступных по ping ip.

есть скриптик, ищите на ntc

**Reply by Keks007** [reply]
На компе заработало сразу после сброса Кеша DNS,а вот с телефона ни через приложение, ни через браузер не хочет, куда рыть?

**Comment by NewUse** [comment]
кажется, достаточно настроить DoH на роутере, отключить там же остальные dns, и почистить кэш днс на конечных клиентах, но сработает далеко не на всех провайдерах. смысла захламлять hosts не вижу, тем более ip там могут меняться со временем суток, ибо это гео-распределённое облако....

**Reply by bol-van** [reply]
Могут, но старые при этом не перестают работать

**Reply by disinformer** [reply]
Настроен DoH на роутере, dnsleaktest.com выдаёт только "правильные" хосты для DNS (dns.adguard-dns.com (Netherlands/Germany)).
Однако получаю ip (157.240.205.174), который от меня не пингуется.
На турецком VPS получаю похожий (157.240.252.174) - пингуется.
На шведском VPS (31.13.72.174) - НЕ пингуется.

Как будто дело случая и удачной DNS-выдачи.

**Reply by numbereleven-a** [reply]
> кажется, достаточно

недостатчно

**Reply by disinformer** [reply]
Поигрался с DNS-провайдерами. Например, у меня такая ситуация по получению ip для instagram.com:

_AdGuard_
Серверы по умолчанию: 157.240.247.174✅
Нефильтрующие серверы: could not find host❌
Семейные серверы: 157.240.205.174❌

_Cloudflare_
Без фильтрации, быстро и конфиденциально: 157.240.0.174✅
Автоматическая защита от вредоносного контента: 31.13.72.174❌
Фильтрация контента для взрослых: 157.240.205.174❌

_Control D_
все кроме профиля "соц сети": 163.70.151.174✅

Вывод: **экспериментируейте с DNS**

**Comment by techurious** [comment]
А как это сделать для винды? Что я делаю не так?
Всё до прописки файла в хост было сделано удачно, только вместо dig пришлось скачать прогу dns lookup. 
- Нашёл рабочий ip через vpn (в моём случае 157.240.0.174)
- добавил его в hosts (157.240.0.174 www.instagram.com)
- Cделал flushdns
- Сайт не открывается (пробовал разные браузеры), хотя пакеты при ping 157.240.0.174 возвращаются.

**Comment by disinformer** [comment]
Забавно, что не было случая, когда бы не работал мобильный клиент (ну с применением zapret).
Не трогают видимо.

**Reply by bol-van** [reply]
К мобильным сетям они подходят более осторожно, боясь навредить экосистеме android

**Comment by scaik** [comment]
есть ли способ подменять ip на правильный с роутера?

**Reply by numbereleven-a** [reply]
> есть ли способ подменять ip на правильный с роутера?

hosts на роутере (openwrt), в целом такой вопрос гуглится вроде достаточно легко

**Reply by scaik** [reply]
разве hosts на роутере не меняет ip только для запросов с роутера? ip запроса определяется на уровне ос клиента, а не роутера. я думал, что можно перенаправлять пакеты instagram.com на другой ip используя zapret. пожалуйста, поправь меня, если я не прав.

**Reply by bol-van** [reply]
dnsmasq обычно читает хосты из /etc/hosts

**Reply by numbereleven-a** [reply]
> разве hosts на роутере не меняет ip только для запросов с роутера? ip запроса определяется на уровне ос клиента, а не роутера. я думал, что можно перенаправлять пакеты instagram.com на другой ip используя zapret. пожалуйста, поправь меня, если я не прав.

если в виндовом hosts у тебя не будет ip для инсты, будет использоваться с роутера, запрет не умеет перенаправлять на другой ip, он не для этого

**Reply by scaik** [reply]
хорошо, спасибо

**Reply by DumbCounselor** [reply]
У меня на openwrt такая история:
 Отключаем использование роутером файла hosts:
`dhcp.@dnsmasq[0].nohosts='1'`
Ну и бинды:
```
dhcp.@domain[0]=domain
dhcp.@domain[0].name='instagram.com'
dhcp.@domain[0].ip='157.240.225.174'
dhcp.@domain[1]=domain
dhcp.@domain[1].name='www.instagram.com'
dhcp.@domain[1].ip='157.240.225.174'
```
Судя по местным инструкциям, потребуется добавить ещё несколько для других доменов.  
Файлы hosts на девайсах и роутере в первозданном виде.

**Comment by th1r93y3** [comment]
привет. а на каком операторе тестили ? у меня на ростелекоме прошло, потом перестал открывать, потом опять 
открыл. как-то вообщем раз через пять удается открыть. при этом пинги проходят исправно.  прописал по третьему варианту используя ip, который через впн (в примере тут такой же), днс не менял. Подскажите плзз, мало опыта в этом деле).  В планах доп роутер и виртуальный сервер, но пока так ...

**Reply by NikittyS** [reply]
Покажи что в hosts прописываешь

**Reply by th1r93y3** [reply]
<img width="463" alt="Screenshot 2025-01-22 at 15 51 08" src="https://github.com/user-attachments/assets/f7bb8558-d96a-4c71-948a-af78d4d59c04" /> 

по graph.instagram.com и  api.instagram.com пинги тоже проходят - их дописывать не стал. 
cdn.instagram.com - не пингуется, даже с впн

**Reply by NikittyS** [reply]
Попробуй так, только пропингуй

157.240.9.174 instagram.com www.instagram.com i.instagram.com z-p42-instagram.c10r.instagram.com
157.240.9.174 scontent-arn2-1.cdninstagram.com scontent.cdninstagram.com static.cdninstagram.com
157.240.22.63 edge-chat.instagram.com instagram.c10r.facebook.com
157.240.22.2 dgw.c10r.facebook.com gateway.instagram.com

**Reply by Nikenet** [reply]
Испытываю похожую проблему.  static.cdninstagram.com с адресом 157.240.9.174 пингуется, но нет ответа на GET запросы

**Reply by th1r93y3** [reply]
не пинговался только z-p42-instagram.c10r.instagram.com , пропинговал с впн, выдал 163.70.128.174, добавил его. Теперь пингуется все из списка. Инста не алле (

**Reply by th1r93y3** [reply]
может этот траффик как-то палиться теперь тоже ? сейчас попробовал ютуб - первый видос проскочил, прогрузился полностью. Дальше - все.  Выглядит похожим образом. Типа dpi не сразу срабатывает, что-то успевает пролететь без ограничений. на уровне догадок.

**Reply by disinformer** [reply]
Какой DNS-провайдер используете? Узнать можно тут: https://www.dnscheck.tools/

Например, у меня - CONTROLD
Получаю следующие ip:
scontent.cdninstagram.com [157.240.205.63] ✅
z-p42-instagram.c10r.instagram.com [57.144.110.34] ✅
instagram.com [57.144.110.34] ✅

В браузере всё открывается.

**Reply by disinformer** [reply]
а вот в процессе экспериментов заметил, что буквально 5-7 мин спустя команда ping возвращает другой ip:
instagram.com [157.240.205.174] ❌
и он уже не пингуется... значит блоченный. сайт соответственно тоже моментами был не алё.

**Reply by th1r93y3** [reply]
Hello! Your public IP addresses are:
ROSTELECOM-MNT
[5.143.183.167](https://info.addr.tools/5.143.183.167)
ptr: 167.net-183.kaluga.ru
Kaluga, Kaluga Oblast, RU
Your DNS resolvers specify your IP subnet ([ECS](https://en.wikipedia.org/wiki/EDNS_Client_Subnet)):
ROSTELECOM-MNT
[5.143.183.0/24](https://info.addr.tools/5.143.183.0/24)
Kaluga, Kaluga Oblast, RU
Your DNS resolvers are:
Google
[74.125.46.18](https://info.addr.tools/74.125.46.18)
ns: ns1.google.com
Hamina, Kymenlaakso, FI
[74.125.46.24](https://info.addr.tools/74.125.46.24)
ns: ns1.google.com
Hamina, Kymenlaakso, FI   **- вот так по dns.**  
 **при пинге айпи не меняется, светится тот же что и прописан, пингуется. через браузер не открывается (**  
 на втором буке прогрузился до логотипа инсты по середине экрана и все. 
 
 поменял ip для инсты на ваш 57.144.110.34 - пингуется, не открывается, загрузился логтип )

**Reply by disinformer** [reply]
Аа... я не уловил, что у вас прописан в hosts. Тогда конечно DNS тут не участвует, вы сами задали ip.

А что по другим "запрещёнкам"? Например, kino.pub или rutracker работают? У меня иногда всё перестаёт работать, тогда перезапускаю службу zapret или весь роутер - всё становится ок.

**Reply by th1r93y3** [reply]
я запрет еще не ставил. жду пока роутерыч приедет. пока попробовал только с инстой по этому примеру.

**Reply by disinformer** [reply]
ёлки-палки... а зачем мозг то сушите заранее))
а мы тут ходим, переживаем за вас... а вас тупо ТСПУ не пускает...

**Reply by NikittyS** [reply]
чел...

**Reply by th1r93y3** [reply]
а что, этот способ самостоятельно не должен работать ?)

**Comment by olshevskyspb** [comment]
Господа, кэширующие сервера Инсты  деградировали по ip + DPI.

**Comment by iamrusskih** [comment]
Хм, добавил следующие домены в список для общего обхода и инста почти полностью работает (кроме директа)

```
instagram.com
cdninstagram.com
igcdn-photos-e-a.akamaihd.net
instagramstatic.com
scontent-hel3-1.cdninstagram.com
static.cdninstagram.com
scontent-arn2-1.cdninstagram.com
scontent.cdninstagram.com
edge-chat.instagram.com
graph.instagram.com
gateway.instagram.com
```
А позже сократил этот список до
```
instagram.com
cdninstagram.com
akamaihd.net
instagramstatic.com
```

**Reply by l00k-1ooK** [reply]
вот пришел к такому же результату. никто не знает как сделать чтоб еще и директ работал?

**Reply by l00k-1ooK** [reply]
решил проблему, написав в hosts эти ip 

157.240.9.174 instagram.com www.instagram.com
157.240.225.174 instagram.com
157.240.225.174 i.instagram.com
157.240.22.2 dgw.c10r.facebook.com
157.240.22.63 instagram.c10r.facebook.com
157.240.22.2 gateway.instagram.com
157.240.22.63 edge-chat.instagram.com
31.13.72.53 scontent-arn2-1.cdninstagram.com scontent.cdninstagram.com
157.240.221.174 instagram.com www.instagram.com
157.240.245.174 instagram.com
157.240.245.174 www.instagram.com
157.240.245.174 b.i.instagram.com
157.240.245.174 z-p42-chat-e2ee-ig.facebook.com
157.240.245.174 help.instagram.com
157.240.205.174 instagram.com www.instagram.com

какой то из них видимо отвечает за директ

**Reply by xMiNiMaLx** [reply]
@l00k-1ooK 
Прописал эти хосты, все они пингуются
но инст по-прежнему недоступен
ощущение, будто 443 порт заблокирован для этого пула адресов (у моего провайдера)

```bash
root@GL-MT6000:~$ ping www.instagram.com
PING www.instagram.com (157.240.0.174): 56 data bytes
64 bytes from 157.240.0.174: seq=0 ttl=56 time=61.307 ms
64 bytes from 157.240.0.174: seq=1 ttl=56 time=61.086 ms
64 bytes from 157.240.0.174: seq=2 ttl=56 time=61.240 ms
64 bytes from 157.240.0.174: seq=3 ttl=56 time=61.028 ms
--- www.instagram.com ping statistics ---
4 packets transmitted, 4 packets received, 0% packet loss
round-trip min/avg/max = 61.028/61.165/61.307 ms

root@GL-MT6000:~$ curl www.instagram.com
^C

root@GL-MT6000:~$ curl -vvv www.instagram.com
^C

root@GL-MT6000:~$ ncat -zv www.instagram.com 443
Ncat: Version 7.95 ( https://nmap.org/ncat )
^C

root@GL-MT6000:~$ ncat -zv ya.ru 443
Ncat: Version 7.95 ( https://nmap.org/ncat )
Ncat: Connected to 77.88.55.242:443.
Ncat: 0 bytes sent, 0 bytes received in 0.03 seconds.
```

**Reply by HentayHentay** [reply]
Переписать в хостс - мало
Это только поможет найти стратегию. 
Ищите

**Reply by xMiNiMaLx** [reply]
> Переписать в хостс - мало Это только поможет найти стратегию. Ищите

Стратегия у меня работает для кучи разных сайтов) а инста никакой стратегией не берётся... с тех пор как поменял провайдера

**Reply by HentayHentay** [reply]
ну тогда остается только редиректить трафик 
запрет потребует много времени для настройки и понимания какие провайдер завалы устроил на пути

**Comment by Grishnakjohn** [comment]
Я добавил диапазоны Ip адресов в список разблокировки и все работает (почти все). Некоторые видео не грузятся, но таких очень мало. При пингах Instagram через VPN понял, что каждый раз Ip разные, но все в указанных ниже диапазонах.
Диапазоны IP 
157.240.0.174-157.240.255.174
31.13.0.174-31.13.255.174 (на счет этих не уверен, но пока оставил.)

**Comment by nilabsent** [comment]
Для обхода заблокированных ip поможет только фиксация ip-hostname через файл hosts. Иначе dns может возвратить ip, который заблокирован, как в случае с инстой, когда то работает то нет. И то это сработает если у сервиса имеются распределённые по миру серверы, а не какой-то один адрес.

Можно резолвить домен через dns, расположенные в разных странах (nslookup не все адреса покажет)
К примеру, такой список dns (надёргал отсюда https://digwebinterface.com ):
```
198.6.100.25
192.76.144.66
195.129.12.122
165.87.13.129
77.88.8.8
94.140.14.14
9.9.9.9
8.26.56.26
144.217.51.168
208.67.222.222
64.6.64.6
168.95.1.1
1.1.1.1
8.8.8.8
```
Проверим instagram.com:
```
nslookup instagram.com 198.6.100.25
157.240.254.174

```
```
nslookup instagram.com 8.8.8.8
157.240.205.174

```
```
nslookup instagram.com 195.129.12.122
185.60.219.174
```
И проверяем полученные ip на доступность 443/TCP посредством, например, nc (через ping проверять бестолку):
```
nc -z -w1 157.240.254.174 443
Connection to 157.240.254.174 443 port [tcp/https] succeeded!
```
Если есть ответ, то с большой вероятностью там есть что-то живое, его и привязываем к домену.
Тоже самое нужно проделать и с cdn (`static.cdninstagram.com` и пр.)
Можно это через скрипт оформить, чтобы проверять сразу множество доменов.
Неработающие домены при загрузке страницы видны через F12
Либо привязывать ip адреса к доменам и направлять запросы к ним через vpn

**Comment by Kesha4765** [comment]
а как сделать все тоже самое, только для FB? Где взять список их серверов (доменные имена)? Надо заставить работать шлем квест 2

**Reply by HentayHentay** [reply]
> а как сделать все тоже самое, только для FB? Где взять список их серверов (доменные имена)?

Через открытые источники (листы FB вроде не раз выкладывались)
Воспользоваться поиском тут и возможно что-то найдем (можно посмотреть в сборках)
К тому-же заблокирован, а значит может потребоваться помимо хостов ещё  и доступный ip (проверяется пингом) с привязыванием через hosts (ip  NS)




> Надо заставить работать шлем квест 2

Тут может быть не протокол сайтов. Так что вам надо будет проснифить .  На основании данной информации написать стратегию или кустом скрипт.

**Reply by Kesha4765** [reply]
а что значит проснифить и главное как?

**Reply by HentayHentay** [reply]
Частично узнаете сами) Что такое  и как это делается. 
Я лишь направляю. 
Никто не отменяет получение доступа с помощью других средств

**Reply by Kesha4765** [reply]
да я примерно понял, надо отследить, к каким серверам стучится шлем и попинговать также их

**Reply by HentayHentay** [reply]
Получение данной информации называется сниффером . Так-же он скажет по каким портам, протоколам, ip и хостам мы обращаемся

**Comment by dead0soul1** [comment]
Ребята с каким приложением вы составляете виртуальный сервер Google DNS , и где вы прописываете команды ?
